src/components/main-area/MainArea.scss

```diff
@@ -1,46 +1,43 @@
-$sidebar-width: 400px;
+@import "../bootstrap/core";
 
+$sidebar-width: 20rem;
 
 .main-area {
-  background: #C4C4C4;
-  height: 93vh;
   position: relative;
   width: 100%;
+  background: $gray-300;
+  height: 93vh;
+}
 
-  .sidebar {
-    position: absolute;
-    left: -$sidebar-width;
-    width: $sidebar-width;
-    border-right: #7e7e7e;
-    border-right-width: 2px;
-    border-right-style: solid;
-    top: 0;
-    bottom: 0;
-    overflow: auto;
-  }
-
-  .sidebar,
-  .pdf-container
-  {
-    transition: left 200ms ease;
-  }
+.sidebar {
+  position: absolute;
+  top: 0;
+  left: -$sidebar-width;
+  bottom: 0;
+  width: $sidebar-width;
+  overflow-x: hidden;
+  overflow-y: auto;
+  background: $gray-200;
+  box-shadow: 0 0 4px rgba(0, 0, 0, 0.25);
+  transition: left 200ms ease;
+  will-change: left;
 
-  .pdf-container {
-    position: absolute;
-    top: 0;
-    right: 0;
-    bottom: 0;
+  .sidebar-open > & {
```

Comment:
What is this selector good for? Could you please explain to me? I don't get it.

Reply:
`&` means current selector and if you explicitly use `&` for this meaning, you can you is in 3 ways.

### Prepend

The following 2 examples are equal:

```scss
ul {
  & li {}
}
```

```scss
ul {
  li {}
}
```

### Append

This is just different way. It allows to say that if this selector is a child of another selector, we apply the properties.

```scss
li {
  ul & {
    padding-left: 3rem;
  }

  ol & {
    padding-left: 4rem;
  }
}
```

The example is kind of stupid. Newer write like this 😉

### Interposition

This is any other way to use the selector.

```scss
li {
  ul &::before {
    content: '-';
  }
}

.active {
  :not(&) {}
}
```

This 2 are not useful but possible.

---

In my case I tie sidebar properties and main area properties to respectively sidebar and main area only but also I add a special condition for the parent element having `.sidebar-open` class name.

Reply:
I didn't know about appending. Thanks for the explanation :)